src/Book.ts

```diff
@@ -0,0 +1,197 @@
+import { Entry } from "./Entry";
+import { IPaginationQuery, IParseQuery, parseQuery } from "./helper/parseQuery";
+import { IJournal, journalModel, TJournalDocument } from "./models/journals";
+import {
+  isValidTransactionKey,
+  ITransaction,
+  transactionModel,
+} from "./models/transactions";
+import type { IOptions } from "./IOptions";
+import type { Document, PipelineStage, Types } from "mongoose";
+
+export class Book<
+  U extends ITransaction = ITransaction,
+  J extends IJournal = IJournal
+> {
+  name: string;
+  precision: number;
+
+  constructor(name: string, options = {} as { precision?: number }) {
+    this.name = name;
+    this.precision =
+      typeof options.precision === "number" ? options.precision : 7;
```

Comment:
Maybe should also add a Number.isInteger check. Maybe makes sense, to throw an Error, as it could have reliability implications.E.g. somebody sets precision to "8" (<--- string) and expects that it will work but it would not have any impact and maybe loses some satoshis. 

Any thoughts?

Reply:
Added validation.